core-services/src/main/java/org/zalando/nakadi/service/timeline/TimelinesFollowerNode.java

```diff
@@ -0,0 +1,157 @@
+package org.zalando.nakadi.service.timeline;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.zookeeper.WatchedEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.service.publishing.NamedThreadFactory;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.Objects;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The class is responsible for managing information about particular timelines node.
+ * The class is acting as a blind follower to zookeeper notifications in regards to locked event types and proposed
+ * configuration version.
+ * <p>
+ * In order to achieve this, this class is reading {@link TimelinesZookeeper#STATE_PATH} of type
+ * {@link VersionedLockedEventTypes} that is used to notify about changes in locked event types and refreshes the
+ * version that this class is compliant with in {@link TimelinesZookeeper#NODES_PATH}/{this_node_id}.
+ */
+@Component
+@Profile("!test")
+public class TimelinesFollowerNode {
+    private final ScheduledExecutorService executorService;
```

Comment:
since Spring Boot is used it is a good practice to schedule tasks on the threads which are managed by framework. `@Scheduled` annotation

Reply:
There are some problems with it - I had to guarantee for 2 different scheduled actions same thread of execution, plus I like the fact that the thread is named